crates/accelerate/src/commutation_checker.rs

```diff
@@ -0,0 +1,742 @@
+    op2: &CircuitInstruction,
+    qargs2: &[u32],
+    cargs2: &[u32],
+    max_num_qubits: u32,
+) -> Option<bool> {
+    if op1.op().control_flow()
+        || op2.op().control_flow()
+        || op1.is_conditioned()
+        || op2.is_conditioned()
+    {
+        return Some(false);
+    }
+
+    // assuming the number of involved qubits to be small, this might be faster than set operations
+    if !qargs1.iter().any(|e| qargs2.contains(e)) && !cargs1.iter().any(|e| cargs2.contains(e)) {
+        return Some(true);
+    }
+
+    if qargs1.len() > max_num_qubits as usize || qargs2.len() > max_num_qubits as usize {
+        return Some(false);
+    }
+
+    if SUPPORTED_OP.contains(&op1.op().name()) && SUPPORTED_OP.contains(&op2.op().name()) {
+        return None;
+    }
+
+    if is_commutation_skipped(op1) || is_commutation_skipped(op2) {
+        return Some(false);
+    }
+
+    None
+}
+
+fn get_op(gate: &PyObject) -> Option<Array2<Complex64>> {
+    Python::with_gil(|py| -> Option<Array2<Complex64>> {
+        Some(
+            QI_OPERATOR
+                .get_bound(py)
+                .call1((gate,))
+                .ok()?
+                .getattr(intern!(py, "data"))
+                .ok()?
+                .extract::<PyReadonlyArray2<Complex64>>()
+                .ok()?
+                .as_array()
+                .to_owned(),
+        )
+    })
+}
+
+fn is_commutation_skipped(instr: &CircuitInstruction) -> bool {
+    let op = instr.op();
+    op.directive() || SKIPPED_NAMES.contains(&op.name()) || instr.is_parameterized()
+}
+
+fn get_relative_placement(
+    first_qargs: &[u32],
+    second_qargs: &[u32],
+) -> SmallVec<[Option<Qubit>; 2]> {
+    let qubits_g2: HashMap<_, _> = second_qargs
+        .iter()
+        .enumerate()
+        .map(|(i_g1, q_g1)| (q_g1, Qubit(i_g1 as u32)))
+        .collect();
+
+    first_qargs
+        .iter()
+        .map(|q_g0| qubits_g2.get(q_g0).copied())
+        .collect()
+}
+
+#[derive(Clone, Debug)]
+#[pyclass]
+pub struct CommutationLibrary {
+    pub library: Option<HashMap<(String, String), CommutationLibraryEntry>>,
+}
+
+impl CommutationLibrary {
+    fn check_commutation_entries(
+        &self,
+        first_op: &OperationRef,
+        first_qargs: &[u32],
+        second_op: &OperationRef,
+        second_qargs: &[u32],
+    ) -> Option<bool> {
+        if let Some(library) = &self.library {
+            match library.get(&(first_op.name().to_string(), second_op.name().to_string())) {
+                Some(CommutationLibraryEntry::Commutes(b)) => Some(*b),
+                Some(CommutationLibraryEntry::QubitMapping(qm)) => qm
+                    .get(&get_relative_placement(first_qargs, second_qargs))
+                    .copied(),
+                _ => None,
+            }
+        } else {
+            None
+        }
+    }
+}
+
+impl ToPyObject for CommutationLibrary {
+    fn to_object(&self, py: Python) -> PyObject {
+        self.library.to_object(py)
+    }
+}
```

Comment:
I get a test failure when doing this and changing line 206 to what you suggested. Any suggestions on how to proceed?
```
  File "/Users/brandhsn/code/qiskit-terra/test/python/circuit/test_commutation_checker.py", line 350, in test_serialization
    cc2 = pickle.loads(pickle.dumps(scc))
                       ^^^^^^^^^^^^^^^^^
TypeError: cannot pickle 'builtins.CommutationLibrary' object
```

Reply:
That just means you need to implement `__getstate__` and `__setstate__`, or `__getnewargs__`, or `__reduce__` depending on your case for `CommutationLibrary`. Since you made it so that it is a `PyClass`, if that's not what you desired and you won't use this struct from `Python`, you could just leave it as a rust struct and leave your implementation of the `ToPyObject` trait, otherwise just implement the correct  serialization method.